### ISSUE TITLE
Transactions not being rolled back properly when upserting items into the inventory during a failure

### ISSUE THREAD

**Issue by webnard** [BYU-ODH/hlrdesk#306]

**Comment by stevethomas19** [comment]
@webnard I am sorry to tag you in so many things, but again not sure what to do here.

**Comment by webnard** [comment]
Yeah, I was pretty obtuse with this. I should have given an example (because who knows if it's even an issue anymore).

I wrote a particularly heinous function called `upsert` in the inventory module that's supposed to either insert an item or update it if it already exists. Postgres, as of January of this year, handles the `UPSERT` operation itself (see http://www.postgresql.org/about/press/presskit95/en/), but our server is still running version 9.1. Might be worth upgrading.

Anyway, long story short, I have no idea what the issue is, or even if it's still around. I committed 1fcd4308 a day after this issue was opened, and it *may* have invalidated this issue. You can play around with it to see if you can cause any sort of bug with transactions (i.e., throw an error right after one of the database calls under the transaction callback and see if it sticks in the database or not [it shouldn't]).

Otherwise, I think this should probably just be closed.

**Comment by stevethomas19** [comment]
> You can play around with it to see if you can cause any sort of bug with transactions (i.e., throw an error right after one of the database calls under the transaction callback and see if it sticks in the database or not [it shouldn't]).

I tested this and it appears the issue was resolved in your commit. I will close the issue.
